src/v2/Apps/Artist/Routes/WorksForSale/Components/ArtistArtworkFilter.tsx

```diff
@@ -37,12 +41,24 @@ const ArtistArtworkFilter: React.FC<ArtistArtworkFilterProps> = props => {
     slug: artist.slug,
   }
 
+  const metric = getSupportedMetric(me?.lengthUnitPreference)
+  const filters: ArtworkFiltersState = {
+    metric,
+    ...match.location.query,
+  }
+
```

Comment:
Does anyone have an idea how best to implement this? Because now it is necessary to add these changes in all places where `ArtworkFilterContextProvider` is used

Reply:
What do you think about making user's preferred metric available in [User](https://github.com/artsy/force/blob/9dc6d876e8c7073aba6f3742585faebea2460bd5/src/v2/System/SystemContext.tsx#L94) from `SystemContext`?

Reply:
I think that makes sense! We would still have to update the different artwork grids, but I think it would just be a matter of adding `const { user } = useSystemContext(); const metric = getSupportedMetric(user.lengthUnitPreference)` which is definitely less steps.

I'm not enough of a Force expert to say for sure if `useSystemContext` is the preferred way of setting and fetching a value like this, but it seems to be widely used at the moment and looks reasonable to me.

Reply:
Similar to @mdole's suggestion, would it be possible to add this as a setting on the settings page? and then that result would be available on the user object. _How_ we get that result is up to us. For example, a hook in the ArtworkFilter:

```tsx
const useUserMetricSetting = () => {
  const { user } = useSystemContext()
  const metric = getSupportedMetric(user.lengthUnitPreference)
  return metric 
}
```

> the user will need to log in again to get a fresh

This is a good question. When the app first boots (meaning, when a user first arrives) we populate the `user` object via `getServerApp` from [here](https://github.com/artsy/force/blob/main/src/v2/System/Router/buildServerApp.tsx#L106) (and corresponding `getClientApp` on the client), which comes from express.js's `req.user`. So once that's loaded on a page request its set, and if anything were to change we'd have to find a way to refresh this object. This is a good problem to solve for that we'll need to eventually get to as force's framework code continues to grow more sophisticated. 

@dzucconi - have we done anything like this before? I can't recall.

Reply:
Another thing about getting the user via `useSystemContext` is I think it would be ok to expand the system context with a `me` prop that makes a `fetchQuery` request out to MP for the `me` field and returns all associated data. Correspondingly, there could be an 
```
const { updateMe } = useSystemContext()
```
which calls a mutation to update `me`. 


I'm not sure exactly what the best architecture for this is off the top of my head, but we use `me` enough within the app to arguably be able to access it through the system context. Note the difference between `me` and `user` -- me has a lot more info, and we've built a lot of UI features on top of it. This wouldn't be a replacement for proper relay-based components that call `me` via a fragment container (we don't want to break Relay's data-masking). 

But that said, the info on me is valuable, and being able to access it globally in this way could be useful.  

This might be a larger discussion worth raising in Slack, however.

Reply:
Slack thread for posterity: https://artsy.slack.com/archives/CP9P4KR35/p1646066609727149

Reply:
@damassi how do you feel about keeping the current implementation of getting the user's preferred metric?

I tested solution that was [discussed yesterday](https://artsy.slack.com/archives/CP9P4KR35/p1646070286694939?thread_ts=1646066609.727149&cid=CP9P4KR35). The main disadvantage is that it will be executed _asynchronously_ and at the time of rendering filters there are no guarantees that the request will be fully executed (we can manually synchronize the values, but is it worth doing so?)
```javascript
export const useUserPreferences = () => {
  const { relayEnvironment } = useSystemContext()
  const [metric, setMetric] = useState<Metric>(DEFAULT_METRIC)

  useEffect(() => {
    const exec = async () => {
      try {
        const response = await fetchQuery<useUserPreferencesQuery>(
          relayEnvironment!,
          graphql`
            query useUserPreferencesQuery {
              me {
                lengthUnitPreference
              }
            }
          `,
          {}
        )

        setMetric(response.me?.lengthUnitPreference)
      } catch (error) {
        console.log("[debug] error", error)
      }
    }

    exec()
  }, [relayEnvironment])

  return { metric }
}
```

Reply:
After our discussion yesterday this occurred to me! That we won't be able to get everything during the SSR pass and then it could cause a lot of staggered UI shifting later. So yeah, totally fine with the current approach if we need it to be synchronous 👍

Reply:
Nice. Thank you!